### PR TITLE
Add tril tests for iudiv, iurem, and ludiv

### PR DESCRIPTION
**Pull request by aviansie-ben**
This PR adds tril tests that exercise the `iudiv`, `iurem`, and `ludiv` opcodes that implement unsigned division and remainder. Additionally, it tweaks the default constants used in tests that make use of unsigned 32- or 64-bit integers to cover more corner cases.

### CODE REVIEW

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Comment by aviansie-ben** [comment]
Opened #3673 for the Power `iudiv`/`iurem` failures. The lack of `ludiv` support on Power is covered by #2227.

**Comment by aviansie-ben** [comment]
Opened #3674 for the Z failure.

File: fvtest/compilertriltest/ArithmeticTest.cpp [hunk 1]
```diff
@@ -109,6 +125,42 @@ TEST_P(Int32Arithmetic, UsingLoadParam) {
...
+    volatile auto exp = param.oracle(param.lhs, param.rhs);
+    volatile auto act = entry_point();
+    ASSERT_EQ(exp, act);
+}
+
+TEST_P(UInt32Arithmetic, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120, "(method return=Int32 args=[Int32, Int32] (block (ireturn (%s (iload parm=0) (iload parm=1)) )))", param.opcode.c_str());
```
Comment (fjeremic) [inline fvtest/compilertriltest/ArithmeticTest.cpp]: I suggest we format these strings so they're easier to read. See example in [1]. Similarly elsewhere in the PR.

[1] https://github.com/eclipse/omr/blob/3e2bef8e9e0840ef3ddb61abc3f2bf5a8b4e0bab/fvtest/compilertriltest/IfxcmpgeReductionTest.cpp#L98-L114

**Comment by aviansie-ben** [comment]
The issue with the Z codegen has been resolved by #3686. As for the Power codegen, I've gone ahead and disabled the tril tests for Power, as it seems like it'll probably be a while before unsigned division is properly supported there.

File: fvtest/compilertriltest/ArithmeticTest.cpp (hunk 1 above)
Comment (aviansie-ben) [reply fvtest/compilertriltest/ArithmeticTest.cpp]: Sorry, I must've somehow missed the notification for this comment. Fixed in 4d37a00315f4bab7ceaae1c567c91baafed7afb5. I also went ahead and formatted the rest of the strings used as templates for trees in `ArithmeticTest.cpp`.

**Comment by fjeremic** [comment]
Going to merge this by EOD unless other committers/reviewers don't have any other comments. Thanks so much for the contribution!

@genie-omr build all

File: fvtest/compilertriltest/ArithmeticTest.cpp [hunk 2]
```diff
@@ -145,6 +259,68 @@ TEST_P(Int64Arithmetic, UsingLoadParam) {
...
+    SKIP_IF(param.opcode == "ludiv" && (OMRPORT_ARCH_PPC64 == arch || OMRPORT_ARCH_PPC64LE == arch), MissingImplementation)
+        << "The Power codegen does not yet support ludiv (see issue #2227)";
+
+    char inputTrees[1024] = {0};
+    std::snprintf(inputTrees, sizeof(inputTrees),
+      "(method return=Int64"
+      "  (block"
+      "    (lreturn"
+      "      (%s"
+      "        (lconst %lld)"
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/ArithmeticTest.cpp]: To avoid causing warnings on different compilers, we're trying to replace these string formats with OMR_PRId*:
```suggestion
      "        (lconst %" OMR_PRId64 ")"
```

Comment (aviansie-ben) [reply fvtest/compilertriltest/ArithmeticTest.cpp]: Fixed in eee6b0e06cc180e296222844971d147b048919ee

**Comment by fjeremic** [comment]
@genie-omr build all
